### DISCUSSION TITLE
Q: Waline 后台无法登录（This request has been blocked; the content must be served over HTTPS.）

### DISCUSSION THREAD

**Discussion by FloatSheep** [walinejs/waline#2290] [Q&A] [answered]
### Describe the bug | 问题描述

Waline 后台无法登录，查看控制台：

Mixed Content: The page at 'https://waline.yurl.eu.org/ui/login?redirect=/ui' was loaded over HTTPS, but requested an insecure resource 'http://waline.yurl.eu.org/api/token?lang=zh-CN'. This request has been blocked; the content must be served over HTTPS.

外网是这样访问的：用户 -> Argo Tunnel -> 雷池（Nginx） -> Waline

Nginx 配置中已经加入了
```
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header REMOTE-HOST $remote_addr;
```

### Website URL | 问题网站

https://waline.yurl.eu.org/ui/

### Where your waline deploy? | 服务部署在哪里？

Other

### Where your comment data store? | 数据存储在哪里？

PostgreSQL

**Comment by lizheming** [comment]
这个就是中间链路没有透传 scheme 协议头的原因，看下是不是 `用户 -> Argo Tunnel` 这块已经变成 http 了呢？这块链路需要你自己查一下了，我也不是特别熟悉你这块的链路。

---
This is the reason why the intermediate link does not transparently transmit the scheme protocol header. Check to see if the `User -> Argo Tunnel` section has become http? You need to check this link yourself. I'm not particularly familiar with your link.

**Reply by FloatSheep** [reply]
从用户这里开始fetch的就是http了，我觉得改一下@waline/admin或许可以解决

---
From the user's point of view, http is the fetch. I think changing @waline/admin may solve the problem

**Reply by lizheming** [reply]
后台的 protocol 是服务端识别请求页面透传过来的 $scheme 决定的

---
The background protocol is determined by the $scheme passed through the server to identify the request page

**Comment by FloatSheep** [comment]
但是 argo 这里是会保持连接协议的，所以我想着试试直接改后台（---- 回复的原邮件 ----发件人Austin ***@***.***>发送日期2024年1月29日 ***@***.***>***@***.***>***@***.***>主题Re: [walinejs/waline] Q: Waline 后台无法登录（This request has been blocked; the content must be served over HTTPS.） (Discussion #2290)后台的 protocol 是服务端识别请求页面透传过来的 $scheme 决定的—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by lizheming** [comment, marked as answer]
`SERVER_URL` 可以通过这个环境变量设置 https://github.com/walinejs/waline/blob/main/packages/server/src/config/extend.js#L15

---
`SERVER_URL` can be set through this environment variable https://github.com/walinejs/waline/blob/main/packages/server/src/config/extend.js#L15
